### ISSUE TITLE
python 3.9: Name node can't be used with 'False' constant

### ISSUE THREAD

**Issue by jamesjer** [sympy/sympy#19638]
Fedora Rawhide is tracking the python 3.9 beta releases.  The latest attempt to build sympy, with python 3.9 beta 3, resulted in a number of test failures, all of which look like this:
```
_____________ sympy/core/tests/test_sympify.py:test_evaluate_false _____________
Traceback (most recent call last):
  File "/builddir/build/BUILD/sympy-sympy-1.6/sympy/core/tests/test_sympify.py", line 435, in test_evaluate_false
    assert sympify(case, evaluate=False) == result
  File "/builddir/build/BUILD/sympy-sympy-1.6/sympy/core/sympify.py", line 406, in sympify
    expr = parse_expr(a, local_dict=locals, transformations=transformations, evaluate=evaluate)
  File "/builddir/build/BUILD/sympy-sympy-1.6/sympy/parsing/sympy_parser.py", line 1006, in parse_expr
    code = compile(evaluateFalse(code), '<string>', 'eval')
ValueError: Name node can't be used with 'False' constant
```
This version of python 3.9 has a function that checks Name nodes and throws an error if the name matches "True", "False", or "None".  My guess is that those should be Constant nodes instead of Name nodes.  I see such Name nodes created, for example, at lines 1061, 1072, 1080, and 1088 of sympy/parsing/sympy_parser.py.

**Comment by jamesjer** [comment]
Indeed, replacing the `ast.Name` expressions on those 4 lines with `ast.Constant(value=False)` led to a successful run of the test suite.  It looks like that should be a correct change for python 3.8 and 3.9, but I don't know about earlier versions.

**Comment by oscarbenjamin** [comment]
Is this the same as #18819?

**Comment by jamesjer** [comment]
I'm not sure if they are the same, but merging https://github.com/sympy/sympy/pull/18819 should fix this issue.

**Comment by asmeurer** [comment]
We should get 3.9 tests passing for the 1.6.1 release, since it will likely be released before 1.7.

**Comment by oscarbenjamin** [comment]
Thanks for reporting this. I've merged #18819 and then the test suite passed in 3.9.
